src/lib/fixer.ts

```diff
@@ -94,10 +96,14 @@ export class _PmdFixGenerator extends FixGenerator {
      * Generate an array of fixes, if possible.
      * @returns
      */
-    public generateFixes(): vscode.CodeAction[] {
+    public generateFixes(processedLines: Set<number>): vscode.CodeAction[] {
         const fixes: vscode.CodeAction[] = [];
         if (this.documentSupportsLineLevelSuppression()) {
-            fixes.push(this.generateLineLevelSuppression());
+            const lineNumber = this.diagnostic.range.start.line;
```

Comment:
This is the main change. Even if we have multiple violations on a single line, we show only one "suppress violation" and that would add the "// NOPMD" tag.